### PR TITLE
feat(web): surface error and warning counts on /ipc recent events card

### PR DESCRIPTION
**Pull request by Peyton-Spencer**
## Summary

Surfaces error and suppressed-event counts inline on the `/ipc` "recent events" stat card so operators can spot a spike of failures without scanning the 50-row timeline below.

The new annotation reuses the existing `1 (2)` parenthetical style from the `retrying` stat card on the same row:

- `50` — no error/warn events in window (unchanged)
- `50 (3 err)` — three error/blocked events present
- `50 (3 err, 2 warn)` — three errors plus two suppressed-style events

The per-row coloring logic that already classifies events into `event-error` / `event-warn` / `event-ok` is now refactored into a shared `classifyIpcEventSeverity` helper so the SSR render, the 5-second live poll in `page-scripts.ts`, and the new rollup count agree on a single classification rule.

Pure render-layer change — no `WebStateProvider`, schema, or SSE event changes.

## Changes

- `src/web/ipc-inspector.ts` — new `classifyIpcEventSeverity`, `countIpcEventSeverities`, and `formatRecentEventsValue` helpers; existing event-row classifier refactored to use them; stat card value now uses the rollup formatter.
- `src/web/page-scripts.ts` — mirrors the helpers in the `/ipc` live poll so the stat card and event-row coloring stay in sync after a poll refresh.
- `src/web/ipc-inspector.test.ts` — direct helper coverage (severity classifier, rollup counter, value formatter) plus integration assertions for the no-issue, error-only, and error+warn rendering branches.
- `docs/observability.md` — documents the new annotation on the `recent events` card in the `/ipc` stat-card table.

## Test plan

- [x] `bun test src/web/ipc-inspector.test.ts` — 44 pass.
- [x] `bun test src/web/page-scripts.test.ts` — 17 pass.
- [x] `bun test` — 2308 pass, 0 fail across the full suite.
- [x] `bun run typecheck` — clean.

Relates to #644 (operator observability rollup tracker).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Enhanced the "recent events" stat card to display error and warning counts inline (e.g., "(X err, Y warn)") alongside the total event count.

* **Documentation**
  * Updated observability reference to clarify the "recent events" card display format and suffix behavior.

* **Tests**
  * Added comprehensive test coverage for event severity classification and display formatting.

### CODE REVIEW

File: src/web/ipc-inspector.ts [hunk 1]
```diff
@@ -22,6 +23,60 @@ const TASK_REASON_CODES = new Set<TaskLaneReason>([
...
+
+/**
+ * Classify an IPC event kind into a severity bucket. Matches the inline
+ * coloring rules used by the event timeline and the live poll script:
+ *
+ * - `error` — anything whose kind mentions `error` or `blocked`
+ * - `warn`  — anything whose kind mentions `suppressed`
+ * - `ok`    — everything else (`message_sent`, `task_created`, etc.)
+ */
+export function classifyIpcEventSeverity(kind: string): IpcEventSeverity {
```
Comment (Peyton-Spencer) [inline src/web/ipc-inspector.ts:42]: Sanity checked this against the current closed IpcEventKind union. The substring classifier is safe for the existing kind set and usefully keeps the SSR stat rollup and row CSS on the same server helper.

File: src/web/page-scripts.ts [hunk 2]
```diff
@@ -1846,13 +1846,15 @@ function ipcScript(): string {
     '  }).catch(function(){});',
     '},5000);',
     '',
+    'function classifyEv(k){k=k||"";return k.indexOf("error")!==-1||k.indexOf("blocked")!==-1?"error":k.indexOf("suppressed")!==-1?"warn":"ok";}',
```
Comment (Peyton-Spencer) [inline src/web/page-scripts.ts:1849]: This mirror matches the server helper semantics and the raw /api/ipc/events array shape. Main future maintenance risk is remembering to update this alongside IpcEventKind/classifyIpcEventSeverity if a new severity-bearing kind is added.

**Review by Peyton-Spencer — COMMENTED** [review]
Review pass for PR #860.

Local verification:
- `bun install --force` passed.
- `bun test src/web/ipc-inspector.test.ts` passed: 44 pass, 0 fail.
- `bun run typecheck` passed.
- Full `bun test` passed: 2308 pass, 0 fail.

Diff verification:
- Reviewed `src/web/ipc-inspector.ts`, `src/web/page-scripts.ts`, `src/web/ipc-inspector.test.ts`, and `docs/observability.md`.
- Server-side SSR and event-row CSS now use the same `classifyIpcEventSeverity()` helper.
- The `recent events` value formatting matches the requested `50 (3 err, 2 warn)` style and omits the parenthetical when there are no error/warn events.
- The inline JS mirror uses the same raw `/api/ipc/events` array shape and the same severity buckets as the server helper.

Risk notes:
- No blocking findings.
- GitHub Actions had not populated in `statusCheckRollup` at review time; CodeRabbit was still pending. Local checks are clean.
- The only maintenance risk is the duplicated inline JS classifier if `IpcEventKind` grows later; I left that note inline.

**Review by Peyton-Spencer — COMMENTED** [review]
Follow-up review for PR #860 after Clayton's sign-off request.

Verification:
- PR #860 is open, non-draft, and mergeable at head `37aa24a582512ca16fb03908f4077147ff0c291c`.
- Earlier clean-checkout verification on this same head passed:
  - `bun install --force`
  - full `bun test`: 2308 pass, 0 fail
  - `bun run typecheck`
- Rechecked focused validation after CodeRabbit completed:
  - `bun test src/web/ipc-inspector.test.ts`: 44 pass, 0 fail
  - `bun run typecheck`: clean

Trace:
- `classifyIpcEventSeverity()` is consistent with the current closed `IpcEventKind` union: `error` / `blocked` -> error, `suppressed` -> warn, otherwise ok.
- `formatRecentEventsValue()` correctly renders a bare total when both error and warn counts are zero, so the parenthetical disappears cleanly for ok-only windows.
- The page-script mirror uses the same severity buckets and the same `fmtEvVal()` formatting, so `50 (3 err, 2 warn)`, warn-only, error-only, and ok-only formatting all line up with SSR when the live table body exists.

Blocking issue before sign-off:
- `src/web/ipc-inspector.ts` renders no `<tbody id="events-body">` when the initial event list is empty; it renders the empty-state div instead.
- In `src/web/page-scripts.ts`, the event poll currently does `var tb=document.getElementById("events-body");if(!tb)return;` before computing counts or updating `#stat-events`.
- That means an `/ipc` page loaded with no events will keep the `recent events` stat stale even after `/api/ipc/events?count=50` starts returning events. This matches CodeRabbit's second actionable and is a real live-refresh bug.

Non-blocking doc nit:
- The docs should also mention the warn-only suffix form `(Y warn)`.

I cannot sign off yet. Fix the live-poll guard order so `#stat-events` updates even when the events table body is absent, and add/adjust a test for the initially-empty event state.

**Comment by Peyton-Spencer** [comment]
Heads up: this PR is missing the CI rollup — only CodeRabbit attached. The other PRs filed today (#859, #864, #865) all show the full 6-check rollup, so this isn't an org-wide outage. Pushing an empty commit to `feat/ipc-events-severity-rollup` should kick the workflows. Possibly related to #484 if the trigger config is workflow-specific. See #866.

**Comment by Peyton-Spencer** [comment]
PM nudge — CI rollup still missing (only CodeRabbit attached), 1 day later. Looks like another instance of the #484 GitHub-App-secret papercut on a non-bot-authored PR. An empty commit (`git commit --allow-empty -m 'ci: re-run'`) usually triggers the workflow. Tracked in #874.
